modular_skyrat/modules/modular_items/code/modular_glasses.dm

```diff
@@ -0,0 +1,297 @@
+
+/obj/item/clothing/glasses/hud/projector/update_icon()
+  if(on)
+    icon_state = initial(icon_state)
+  else
+    icon_state = off_state
+
+/obj/item/clothing/glasses/hud/projector/attack_self(mob/living/user)
+  if(toggleable)
+    if(on)
+      to_chat(usr, span_notice("You deactivate the optical meson matrix on the [src]."))
+      on = FALSE
+      icon_state = off_state
+      user.update_inv_glasses()
+      flash_protect = FLASH_PROTECTION_NONE
+      vision_flags = 0
+      hud_type = null
+      hud_trait = null
+      tint = 0
+      clothing_traits = null
+    else
+      to_chat(usr, span_notice("You activate the optical meson matrix on the [src]."))
+      on = TRUE
+      icon_state = initial(icon_state)
+      user.update_inv_glasses()
+      flash_protect = initial(flash_protect)
+      tint = initial(tint)
+      vision_flags = initial(vision_flags)
+      hud_type = initial(hud_type)
+      hud_trait = initial(hud_trait)
+      clothing_traits = initial(clothing_traits)
+    update_icon()
+    playsound(src, activation_sound, 50, TRUE)
+    user.update_inv_glasses()
+    user.update_action_buttons()
+    user.update_sight()
+  ..()
+
+/obj/item/clothing/glasses/hud/projector/meson
+  name = "retinal projector meson HUD"
+  desc = "A headset equipped with a scanning lens and mounted retinal projector. It doesn't provide any eye protection, but it's less obtrusive than goggles."
+  icon_state = "projector_meson"
+  lighting_alpha = 300
+  vision_flags = SEE_TURFS 
+  darkness_view = 2
+
+/obj/item/clothing/glasses/hud/projector/health
+  name = "retinal projector health HUD"
+  desc = "A headset equipped with a scanning lens and mounted retinal projector. It doesn't provide any eye protection, but it's less obtrusive than a visor."
+  icon_state = "projector_med"
+  hud_type = DATA_HUD_MEDICAL_ADVANCED
+  hud_trait = TRAIT_MEDICAL_HUD
+
+/obj/item/clothing/glasses/hud/projector/security
+  name = "retinal projector security HUD"
+  desc = "A headset equipped with a scanning lens and mounted retinal projector. It doesn't provide any eye protection, but it's less obtrusive than a visor."
+  icon_state = "projector_sec"
+  hud_type = DATA_HUD_SECURITY_ADVANCED
+  hud_trait = TRAIT_SECURITY_HUD
+
+/obj/item/clothing/glasses/hud/projector/diagnostic
+  name = "retinal projector diagnostic HUD"
+  desc = "A headset equipped with a scanning lens and mounted retinal projector. It doesn't provide any eye protection, but it's less obtrusive than a visor."
+  icon_state = "projector_diagnostic"
+  hud_type = DATA_HUD_DIAGNOSTIC_BASIC
+  hud_trait = TRAIT_DIAGNOSTIC_HUD
+
+/obj/item/clothing/glasses/hud/projector/science
+  name = "science retinal projector"
+  desc = "A headset equipped with a scanning lens and mounted retinal projector. It doesn't provide any eye protection, but it's less obtrusive than a visor."
+  icon_state = "projector_sci"
+  clothing_traits = list(TRAIT_REAGENT_SCANNER, TRAIT_RESEARCH_SCANNER)
+
+// Designs
```

Comment:
You might want to move these to a separate research design file. One might already exist for glasses, assuming that the prescription HUDs on the techweb are a skyrat change.